### PR TITLE
Add sulu_category_url_remove and sulu_category_url_toggle twig functions

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | -
| License | MIT
| Documentation PR | sulu/sulu-docs# 

#### What's in this PR?

Add `sulu_category_url_remove` and `sulu_category_url_toggle` twig functions

#### Why?

Because there is just a `sulu_category_url_append` function and no way to remove a category from the url

#### To Do

- [ ] Create a documentation PR

### CODE REVIEW

File: src/Sulu/Component/Category/Tests/Unit/Request/CategoryRequestHandlerTest.php [hunk 1]
```diff
@@ -86,6 +86,78 @@ public function testAppendToUrl($parameter, $url, $queryString, $expected)
...
+            ['categories', '/asdf', '1,2', '1,2'],
+            ['categories', '/test', '1,3,2', '1,2'],
+            ['categories', '/test', '3,1', '1'],
+            ['categories', '/test', '1,3', '1'],
+            ['categories', '/test', '', ''],
+        ];
+    }
+
+    /**
+     * @dataProvider removeProvider
```
Comment (luca-rath) [inline src/Sulu/Component/Category/Tests/Unit/Request/CategoryRequestHandlerTest.php]:
```suggestion
    public function removeSingleProvider()
    {
        return [
            ['c', '/test', '1,2,3', '1,2'],
            ['c', '/asdf', '1,2', '1,2'],
            ['c', '/asdf', '3', ''],
            ['categories', '/asdf', '1,2', '1,2'],
            ['categories', '/test', '1,3,2', '1,2'],
            ['categories', '/test', '3,1', '1'],
            ['categories', '/test', '1,3', '1'],
            ['categories', '/test', '', ''],
        ];
    }

    /**
     * @dataProvider removeSingleProvider
```

File: src/Sulu/Component/Category/Tests/Unit/Request/CategoryRequestHandlerTest.php [hunk 2]
```diff
@@ -86,6 +86,78 @@ public function testAppendToUrl($parameter, $url, $queryString, $expected)
...
+            ['categories', '/test', '3,1', '1'],
+            ['categories', '/test', '1,3', '1'],
+            ['categories', '/test', '', ''],
+        ];
+    }
+
+    /**
+     * @dataProvider removeSingleProvider
+     */
+    public function testRemoveToUrl($parameter, $url, $queryString, $expected)
```
Comment (luca-rath) [inline src/Sulu/Component/Category/Tests/Unit/Request/CategoryRequestHandlerTest.php]:
```suggestion
    public function testRemoveSingleFromUrl($parameter, $url, $queryString, $expected)
```

**Comment by alexander-schranz** [comment]
@luca-rath can you rebase this on 2.1 branch?

**Comment by alexander-schranz** [comment]
@luca-rath we need also add documentation page for them and maybe link between this similar twig extension in the docs.

**Pull request merged**